ops/testing.py

```diff
@@ -458,6 +485,63 @@ def add_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
         self._charm.on[relation_name].relation_joined.emit(
             relation, remote_unit.app, remote_unit)
 
+    def remove_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
+        """Remove a unit from a relation.
+
+        Example::
+
+          rel_id = harness.add_relation('db', 'postgresql')
+          harness.add_relation_unit(rel_id, 'postgresql/0')
+          ...
+          harness.remove_relation_unit(rel_id, 'postgresql/0')
+
+        This will trigger a `relation_departed` event. This would
+        normally be followed by a `relation_changed` event triggered
+        by Juju. However when using the test harness a
+        `relation_changed` event must be triggererd using
+        :meth:`.update_relation_data`. This deviation from normal Juju
+        behaviour, facilitates testing by making each step in the
+        charm life cycle explicit.
+
+        Args:
+            relation_id: The integer relation identifier (as returned by add_relation).
+            remote_unit_name: A string representing the remote unit that is being added.
+
+        Raises:
+            KeyError: if relation_id or remote_unit_name is not valid
+            ValueError: if remote_unit_name is not valid
+        """
+        relation_name = self._backend._relation_names[relation_id]
+
+        # gather data to invalidate cache later
+        if self._model is not None:  # defensive check for model
+            remote_unit = self._model.get_unit(remote_unit_name)
+            relation = self._model.get_relation(relation_name, relation_id)
+            unit_cache = relation.data.get(remote_unit, None)
+
+        # statements which could access cache
+        self._emit_relation_departed(relation_id, remote_unit_name)
+        self._backend._relation_data[relation_id].pop(remote_unit_name)
+        self._backend._relation_app_and_units[relation_id][
+            "units"].remove(remote_unit_name)
+        self._backend._relation_list_map[relation_id].remove(remote_unit_name)
+
+        if unit_cache is not None:
+            unit_cache._invalidate()
```

Comment:
Isn't a better way for doing this than calling a private method?

Reply:
This seems to be the practice adopted elsewhere. See for instance `add_relation_unit`. Is there an alternative you would suggest ?

Reply:
Harness as the way to mutate the Model without actually exposing ways to mutate Model directly was decided as the best trade off to have testability but not runtime mutability.